src/ucp/core/ucp_worker.c

```diff
@@ -164,18 +164,22 @@ static void ucp_worker_set_am_handlers(ucp_worker_iface_t *wiface, int is_proxy)
 
     ucs_trace_func("iface=%p is_proxy=%d", wiface->iface, is_proxy);
 
-    for (am_id = 0; am_id < UCP_AM_ID_LAST; ++am_id) {
+    for (am_id = 1; am_id < UCP_AM_ID_LAST; ++am_id) {
         if (!(wiface->attr.cap.flags & (UCT_IFACE_FLAG_AM_SHORT |
                                         UCT_IFACE_FLAG_AM_BCOPY |
                                         UCT_IFACE_FLAG_AM_ZCOPY))) {
             continue;
         }
 
-        if (!(context->config.features & ucp_am_handlers[am_id].features)) {
+        if (!ucp_am_handlers[am_id]) {
```

Comment:
why needed?

Reply:
> why needed?

am_id=5 is not defined - caused segfault

Reply:
ucp_am_handlers[am_id] != NULL

Reply:
> yes, `ucp_am_handlers[am_id] == NULL` (seems not fixed)

missed in unstashed. fixed